JAICore/jaicore-ml/src/jaicore/ml/core/dataset/IDataset.java

```diff
@@ -6,18 +6,48 @@
 
 import jaicore.ml.core.dataset.attribute.IAttributeType;
 
-public interface IDataset {
-
-	public IAttributeType getTargetType();
-
-	public List<IAttributeType> getAttributeTypes();
-
-	public List<IInstance> getInstances();
-
-	public IInstance get(int index);
-
+/**
+ * Common interface of a dataset defining methods to access meta-data and instances contained in the dataset. Moreover, methods for serialization and deserialization are defined.
+ *
+ * @author wever
+ */
+public interface IDataset extends List<IInstance> {
+
+	/**
+	 * Returns the attribute type of the target attribute.
+	 *
+	 * @return The attribute type of the target attribute.
+	 */
+	public IAttributeType<?> getTargetType();

```

Comment:
I would rather choose the same trick we did in IInstance by parameterizing the getter with a generic. The <?> is pretty useless in my opinion, because you have no idea what it is.